db/version_set.cc

```diff
@@ -1865,6 +1925,18 @@ void Version::Get(const ReadOptions& read_options, const LookupKey& k,
         // TODO: update per-level perfcontext user_key_return_count for kMerge
         break;
       case GetContext::kFound:
+        if (is_blob_index) {
+          if (do_merge && value) {
```

Comment:
Yes, I think it does: `do_merge` is actually a misnomer since it's supposed to mean "get value" (it's what differentiates the `Get` and `GetMergeOperands` APIs). From `version_set.h`:

```
  // Lookup the value for key or get all merge operands for key.
  // If do_merge = true (default) then lookup value for key.
```